haystack/document_stores/memory.py

```diff
@@ -281,47 +281,57 @@ def get_document_by_id(
         else:
             return None
 
-    def get_documents_by_id(self, ids: List[str], index: Optional[str] = None) -> List[Document]:  # type: ignore
+    def get_documents_by_id(
+        self,
+        ids: List[str],
+        index: Optional[str] = None,
+        batch_size: Optional[int] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> List[Document]:
         """
         Fetch documents by specifying a list of text id strings.
         """
+        if headers:
+            raise NotImplementedError("InMemoryDocumentStore does not support headers.")
+        if batch_size:
+            logger.warning(
+                "InMemoryDocumentStore does not support batching in `get_documents_by_id` method. This parameter is ignored."
+            )
         index = index or self.index
         documents = [self.indexes[index][id] for id in ids]
         return documents
 
-    def get_scores_torch(self, query_emb: np.ndarray, document_to_search: List[Document]) -> List[float]:
+    def get_scores_torch(self, query_emb: np.ndarray, documents_to_search: List[Document]) -> List[float]:
         """
         Calculate similarity scores between query embedding and a list of documents using torch.
 
         :param query_emb: Embedding of the query (e.g. gathered from DPR)
-        :param document_to_search: List of documents to compare `query_emb` against.
+        :param documents_to_search: List of documents to compare `query_emb` against.
         """
-        query_emb = torch.tensor(query_emb, dtype=torch.float).to(self.main_device)  # type: ignore [assignment]
-        if len(query_emb.shape) == 1:
-            query_emb = query_emb.unsqueeze(dim=0)  # type: ignore [attr-defined]
+        query_emb_tensor = torch.tensor(query_emb, dtype=torch.float).to(self.main_device)
+        if len(query_emb_tensor.shape) == 1:
+            query_emb_tensor = query_emb_tensor.unsqueeze(dim=0)
 
-        doc_embeds = np.array([doc.embedding for doc in document_to_search])
-        doc_embeds = torch.as_tensor(doc_embeds, dtype=torch.float)  # type: ignore [assignment]
-        if len(doc_embeds.shape) == 1 and doc_embeds.shape[0] == 1:
-            doc_embeds = doc_embeds.unsqueeze(dim=0)  # type: ignore [attr-defined]
-        elif len(doc_embeds.shape) == 1 and doc_embeds.shape[0] == 0:
+        doc_embeds = np.array([doc.embedding for doc in documents_to_search])
+        doc_embeds_tensor = torch.as_tensor(doc_embeds, dtype=torch.float)
+        if len(doc_embeds_tensor.shape) == 1 and doc_embeds_tensor.shape[0] == 0:
```

Comment:
`doc_embeds_tensor.shape[0]` is equivalent to `len(doc_embeds_tensor)` here, correct ([numpy docs](https://numpy.org/devdocs/reference/generated/numpy.shape.html))?
What if `len(doc_embeds_tensor.shape) == 1 and doc_embeds_tensor.shape[0] == 1`? In the past, we then called `unsqueeze(dim=0)`. Do I understand correctly that we don't need that step and can continue with the execution?
Maybe we can add a comment to the `if len(doc_embeds_tensor.shape) == 1 and doc_embeds_tensor.shape[0] == 0:` then. Explaining that no scores are returned if none of the provided documents has an embedding? That would make the code easier to read. What do you think?

Reply:
Hello @julian-risch!
Thanks for the review and sorry for not being understandable. 🙂 

If you look at https://github.com/deepset-ai/haystack/issues/4080#issuecomment-1423152835, what emerges is that the original code was a bit buggy. I did a too quick fix, which was not very clear and maybe not so correct.

Now I tried to write more expressive code: for example, I used `ndim` instead of `len(array.shape)`.
I also reintegrated that `unsqueeze`/`expand_dims` part (though I think that it will probably be unused) in a more reasonable way, I hope.

Now my code needs another careful review, also because I'm not a tensor wizard 🧙‍♂️.

Reply:
There is also a new test failing: `Add label on docstrings edit / label`
It's not related to this PR, I think.